packages/pn-validator/src/Validator.ts

```diff
@@ -4,12 +4,39 @@ import { ValidatorBuilder } from './ValidatorBuilder';
 import { hasError } from './HasError';
 import { TypeRules } from './types/TypeRules';
 
+/**
+ * Validator options that include the strict mode but in the future you can add any other
+ * options.
+ */
+export interface ValidatorOptions { 
+  strict?: boolean;
+}
+
 export class Validator<TModel> {
   private validatorBuilders: ValidatorBuilders<TModel> = {};
+  private strict: boolean; // strict mode
+
+  constructor(options: ValidatorOptions = {}) {
+    this.strict = options.strict ?? false;
+  }
 
   public readonly validate = (model: TModel): ValidationError<TModel> | null => {
+
+    // check if the number of rule keys are equals to the number of model keys.
+    if (this.strict) {
+      if (typeof model === 'object' && model !== null) {
+        const modelKeys = Object.keys(model);
+        const ruleKeys = Object.keys(this.validatorBuilders);
+        const missingKeys = modelKeys.filter((key) => !ruleKeys.includes(key));
+
+        if (missingKeys.length > 0) {
```

Comment:
I shouldn't return an error but a Validation error as done for the others errors

Reply:
fixed in the commit: 495f4721b9553d2ad535be08bc57d7c7aee30be9